ocean_lib/assets/test/test_asset_downloader.py

```diff
@@ -5,98 +5,139 @@
             publisher_wallet,
             "test_destination",
-            "",
             "test_order_tx_id",
-            data_provider,
-            index,
+            index=index,
         )
+
     index = 4
-    with pytest.raises(AssertionError):
+    with pytest.raises(Exception):
         download_asset_files(
-            sa.index,
             ddo,
+            config.provider_url,
             publisher_wallet,
             "test_destination",
-            "",
             "test_order_tx_id",
-            data_provider,
-            index,
+            index=index,
         )
 
 
-def test_ocean_assets_download_destination_file(tmpdir, publisher_wallet):
+def test_ocean_assets_download_destination_file(
+    web3, config, tmpdir, publisher_wallet, publisher_ocean_instance
+):
     """Convert tmpdir: py._path.local.LocalPath to str, satisfy enforce-typing."""
-    ocean_assets_download_destination_file_helper(str(tmpdir), publisher_wallet)
+    ocean_assets_download_destination_file_helper(
+        web3, config, str(tmpdir), publisher_wallet, publisher_ocean_instance
+    )
 
 
-def ocean_assets_download_destination_file_helper(tmpdir, publisher_wallet):
+def ocean_assets_download_destination_file_helper(
+    web3, config, tmpdir, publisher_wallet, publisher_ocean_instance
+):
     """Tests downloading to an existing directory."""
     data_provider = DataServiceProvider
+    erc721_token, erc20_token = deploy_erc721_erc20(
+        web3, config, publisher_wallet, publisher_wallet, cap=to_wei(100)
+    )
+
+    _, metadata, encrypted_files = create_basics(config, web3, data_provider)
+    ddo = publisher_ocean_instance.assets.create(
+        metadata=metadata,
+        publisher_wallet=publisher_wallet,
+        encrypted_files=encrypted_files,
+        erc721_address=erc721_token.address,
+        deployed_erc20_tokens=[erc20_token],
+    )
+    access_service = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+
+    erc20_token.mint(
+        account_address=publisher_wallet.address,
+        value=to_wei(50),
+        from_wallet=publisher_wallet,
+    )
 
-    ddo = get_sample_ddo()
-    sa = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+    initialize_response = data_provider.initialize(
+        did=ddo.did,
+        service_id=access_service.id,
+        consumer_address=publisher_wallet.address,
+        service_endpoint=data_provider.build_initialize_endpoint(config.provider_url)[
+            1
+        ],
+    )
+
+    tx_id = erc20_token.start_order(
+        consumer=publisher_wallet.address,
+        service_id=int(access_service.id),
+        provider_fees=initialize_response.json()["providerFee"],
+        from_wallet=publisher_wallet,
+    )
 
     written_path = download_asset_files(
-        sa.index,
-        ddo,
-        publisher_wallet,
-        tmpdir,
-        "0x1",
-        "test_order_tx_id",
-        data_provider,
+        ddo, config.provider_url, publisher_wallet, tmpdir, tx_id
```

Comment:
what is the difference between this test and the consume flow? I would rather keep the tests here, but please make sure you assert more than the path existence, rather copy the part with len(dir) here.